text/0117-typo-tolerance-settings-api.md

```diff
@@ -0,0 +1,228 @@
+- Title: Typo Tolerance Settings API
+- Start Date: 2022-02-22
+
+# Typo Tolerance API
+
+## 1. Summary
+
+This specification describes the customizable options for the typo tolerance settings.
+
+## 2. Motivation
+
+Tolerance to typos is critical when building an enjoyable search experience. Meilisearch ranks typo-free matched documents above others based on the default settings that we believe are best for most users.
+
+Meilisearch can adapt to different use-cases by providing customization options for the typo tolerance feature.
+
+## 3. Functional Specification
+
+### 3.1. `typoTolerance` API resource definition
+
+| Field                                            | Type            | Required |
+|--------------------------------------------------|-----------------|----------|
+| [enabled](#311-enabled)                          | Boolean         | False    |
+| [disableOnAttributes](#312-disableonattributes)  | Array of String | False    |
+| [disableOnWords](#313-disableonwords)            | Array of String | False    |
+| [minWordSizeFor1Typo](#314-minwordsizefor1typo)  | Integer         | False    |
+| [minWordSizeFor2Typos](#315-minwordsizefor2typos)| Integer         | False    |
+
+#### 3.1.1. `enabled`
+
+- Type: Boolean
+- Required: False
+- Default: `true`
+
+Whether the typo tolerance feature is enabled.
+
+##### 3.1.1.1. Impacts on the `typo` ranking rule
+
+The presence of `typo` in the ranking rules setting does not influence the activation/deactivation of the typo tolerance feature.
+
+If the `rankingRules` parameter of the index settings does not contain the `typo` rule, the results are not sorted according to the number of typos found.
+
+The typo tolerance feature is still used to match documents.
+
+### 3.1.2. `disableOnAttributes`
+
+- Type: Array of String
+- Required: False
+- Default: `[]`
+
+`disableOnAttributes` disable the typo tolerance feature on the specified attributes list.
+
+Any attributes defined in `disableOnAttributes` won't have their values matched by the typo tolerance.
```

Comment:
this part is really ambitious, needing changes in both search and indexing parts.
Moreover, it could create conflict with other works on the indexer:
- nested fields
- optimizations

I suggest making a discovery on this specific part.

Reply:
@MarinPostma If it helps!